android/runtime/common/src/js/console.js

```diff
@@ -44,3 +45,28 @@ exports.error = function () {
 exports.debug = function () {
 	Titanium.API.debug(join(arguments));
 };
+
+exports.time = function (label) {
+	if (times[label]) {
+		exports.warn('Label "' + label + '" already exists');
+		return;
+	}
+	if (!label) {
+		label = 'default';
+	}
+	times[label] = Date.now();
+};
+
+exports.timeEnd = function (label) {
+	if (!label) {
+		label = 'default';
+	}
+	var startTime = times[label];
```

Comment:
vars should be declared at top of scope, since they get hoisted up there.

Reply:
Also, given that we know V8 is newer and can support ES6 natively, is there any reason we can't start making use of es6 syntax in runtime JS code?  (i.e. template strings here, use of const/let)

Reply:
e.g. https://github.com/nodejs/node/blob/master/lib/console.js#L166-L184

Reply:
@sgtcoolguy  Updated to use template literals, const and a Map instead of a plain object

Reply:
FYI I filed https://jira.appcelerator.org/browse/TIMOB-25892 to hook the linting up in this folder as `grunt lint` doesn't include it currently